packages/ui/src/components/branch-selector/branch-selector-dropdown.tsx

```diff
@@ -0,0 +1,189 @@
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+
+import {
+  Badge,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  Icon,
+  SearchBox,
+  Tabs,
+  TabsList,
+  TabsTrigger,
+  Text
+} from '@/components'
+import { cn } from '@utils/cn'
+
+import { BranchSelectorBranchProps, BranchSelectorItem, BranchSelectorTab } from './types'
+
+const BRANCH_SELECTOR_LABELS = {
+  [BranchSelectorTab.BRANCHES]: {
+    label: 'Branches',
+    searchPlaceholder: 'Find a branch'
+  },
+  [BranchSelectorTab.TAGS]: {
+    label: 'Tags',
+    searchPlaceholder: 'Find a tag'
+  }
+} as const
+
+export interface BranchSelectorDropdownProps {
+  name: string
+  branchList: BranchSelectorItem[]
+  tagList?: BranchSelectorItem[]
+  selectBranch: (branch: string) => void
+}
+
+const filterItems = (items: BranchSelectorBranchProps[] | BranchSelectorItem[], query: string) => {
+  if (!query.trim()) return items
+
+  return items.filter(item => item.name.toLowerCase().includes(query.toLowerCase().trim()))
+}
+
+export const BranchSelectorDropdown = ({ name, branchList, tagList, selectBranch }: BranchSelectorDropdownProps) => {
+  const [activeTab, setActiveTab] = useState<BranchSelectorTab>(BranchSelectorTab.BRANCHES)
+  const [searchQuery, setSearchQuery] = useState('')
+
+  const sourceItems = activeTab === BranchSelectorTab.BRANCHES ? branchList : (tagList ?? [])
+  const filteredItems = filterItems(sourceItems, searchQuery)
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value)
+  }
+
+  const viewAllUrl = activeTab === BranchSelectorTab.BRANCHES ? '' : ''
+
+  return (
+    <DropdownMenuContent className="w-[298px] p-0" align="start">
+      <div className="px-3 pt-2">
+        <Text className="leading-none" size={2} weight="medium">
+          Switch branches/tags
+        </Text>
+
+        <SearchBox.Root
+          className="mt-[18px] w-full"
+          placeholder={BRANCH_SELECTOR_LABELS[activeTab].searchPlaceholder}
+          value={searchQuery}
+          handleChange={handleSearchChange}
+          showOnFocus
+        />
+      </div>
+
+      <Tabs
+        className="mt-2"
+        variant="branch"
+        value={activeTab}
+        onValueChange={value => {
+          setActiveTab(value as BranchSelectorTab)
```

Comment:
Not one for now, but we should allow a type to be set for what the tabs are so we can get some type safety instead of just a string. Would have been nice to do...

```typescript
<Tabs<BranchSelectorTab>
  onValueChange={value => {
    setActiveTab(value)
```

One for another day